src/main/java/hudson/plugins/analysis/views/DetailFactory.java

```diff
@@ -199,7 +200,7 @@ else if (link.startsWith("type.")) {
      *            the default encoding to be used when reading and parsing files
      * @return the detail view
      */
-    protected AttributeDetail createAttributeDetail(final AbstractBuild<?, ?> owner, final DefaultAnnotationContainer annotations,
+    protected AttributeDetail createAttributeDetail(final Run<?, ?> owner, final DefaultAnnotationContainer annotations,
```

Comment:
I like this type of comments, really constructive. Thanks!
Added compatibility method at the end of the class.